explainer.md

```diff
@@ -60,18 +60,20 @@ The entire painting of the element is captured, including things which appear ou
 
 https://user-images.githubusercontent.com/93594/141118353-d62d19a1-0964-4fa0-880f-bdde656ce899.mp4
 
-The element is captured without the effects (such as opacity and filters) from parent elements. Effects on the element itself are baked into the image. However, the element is captured without transforms, as those transforms are reapplied later.
+The element is captured without the effects (such as opacity and filters) from parent elements. Effects on the element itself are baked into the image. However, the element is captured without transforms, as those transforms are reapplied later. The root is always captured as a single image, with the other transition elements removed (similar to how compositing works today).
 
 Capturing an element in this way isn't a new concept to the platform, as [`element()`](<https://developer.mozilla.org/en-US/docs/Web/CSS/element()>) in CSS performs a similar action. The differences are documented later.
 
-The root is always captured as a single image, with the other transition elements removed (similar to how compositing works today), and is also clipped to the viewport, as capturing the entire page would take an enormous amount of memory in many cases.
-
 Capturing as a CSS image avoids the interactivity risks, complexities, and memory impact of fully preserving these parts of Page-A as live DOM. On the other hand, it means that the capture will be 'static'. If it includes things like gifs, video, or other animating content, they'll be frozen on the frame they were displaying when captured.
 
-- [Open question](https://github.com/WICG/shared-element-transitions/issues/72): Should we have a way to expand the root capture area for particular transitions? For example, transitions that involve vertical movement?
-- [Open question](https://github.com/WICG/shared-element-transitions/issues/73): Other elements can also be massive. Do we need a way to limit and control the captured size of those?
+#### Image Size
+The size of the image cached for an element is equal to its border box size + the ink overflow from visual effects like shadows and blurs applied to the element. This allows exposing parts of an element during the transition which may have been hidden earlier. If an element is massive a UA may need to clip the image to a capped size (a common case would be the max texture size supported by the device). When caching a subset of the element due to this constraint, the rect within the element cached by the UA is the area closest to the viewport (computed using a heuristic).
```

Comment:
```suggestion
The size of the image cached for an element is equal to the element's [ink overflow rectangle](https://drafts.csswg.org/css-overflow-3/#ink-overflow-rectangle). This allows exposing parts of an element during the transition which may have been hidden earlier. The user-agent is allowed to clip the image to an implementation defined size (a common case would be the max texture size supported by the device). When caching a subset of the element due to this constraint, the area within the element cached by the user-agent is the area closest to the viewport.
```

(I _think_ what you're describing is the ink overflow rectangle, but please verify)

Reply:
I don't think it's ink overflow since the text there mentions : "of a box and its contents". I'm assuming contents implies overflow from the element's descendents in addition to the element's own visual effects. While we only mean overflow from the element's own effects.

Reply:
Discussed offline. Saying ink overflow rect here is fine since paint containment already implies that overflow from the content (or descendents) will be clipped.

This does imply that we can have overflow due to overflow-clip-margin and we assumed that the image size doesn't need to expand for that. But we can resolve that detail later.